clang/test/ClangScanDeps/error.cpp

```diff
@@ -1,23 +1,10 @@
 // RUN: rm -rf %t
 // RUN: split-file %s %t
 
-//--- missing_tu.json.in
-[{
-  "directory": "DIR",
-  "command": "clang -fsyntax-only DIR/missing_tu.c",
-  "file": "DIR/missing_tu.c"
-}]
-//--- missing_header.json.in
-[{
-  "directory": "DIR",
-  "command": "clang -fsyntax-only DIR/missing_header.c",
-  "file": "DIR/missing_header.c"
-}]
 //--- missing_header.c
 #include "missing.h"
 
-// RUN: sed -e "s|DIR|%/t|g" %t/missing_tu.json.in > %t/missing_tu.json
-// RUN: not clang-scan-deps -compilation-database %t/missing_tu.json 2>%t/missing_tu.errs
+// RUN: not clang-scan-deps -- "clang" %t/missing_tu.c 2>%t/missing_tu.errs
```

Comment:
what's up with the quoting around "clang"?

Also, why was -fsyntax-only removed?

Reply:
LIT complains about unquoted `clang` in attempt to nudge people towards using `%clang`. That, however, adds a bunch of default flags to the invocation and I'm not sure I want those in our scanner tests (especially when testing the search path optimization for example).

The "-fsyntax-only" flag was removed because it's unnecessary (it was originally used to prevent the scanner actually generating compilation output files I think) and because it implies no output file (which the per-file mode currently requires). Since the scanner is intended to be run on actual compilation jobs, I think it's even desirable removing stuff like "-fsyntax-only".

Reply:
I just tried double-checking what `%clang` expands to and I was wrong, it just expands to the Clang binary in the build directory, so I think I can remove the quote workaround.

However, both `"clang"` and `%clang` expand to `-cc1` invocations that refer to system directories: `-isystem /usr/local/include -internal-externc-isystem /usr/include`. It would be nice to have something that cleans these out (like `%clang_cc1` does), but I consider that out of scope for this patch. (Not functional change from in this aspect anyways.)